xds/src/test/java/io/grpc/xds/internal/sds/XdsChannelBuilderTest.java

```diff
@@ -35,7 +37,19 @@ public void buildsXdsChannelBuilder() {
     XdsChannelBuilder builder = XdsChannelBuilder.forTarget("localhost:8080");
     assertThat(builder).isNotNull();
     assertThat(builder.delegate()).isInstanceOf(NettyChannelBuilder.class);
+    builder = builder.fallbackProtocolNegotiator(InternalProtocolNegotiators.plaintext());
     ManagedChannel channel = builder.build();
     assertThat(channel).isNotNull();
   }
+
+  @Test
+  public void buildsXdsChannelBuilder_noFallbackNegotiator() {
+    XdsChannelBuilder builder = XdsChannelBuilder.forTarget("localhost:8080");
+    try {
+      builder.build();
+      Assert.fail("no exception thrown");
+    } catch (NullPointerException expected) {
+      assertThat(expected).hasMessageThat().isEqualTo("fallbackProtocolNegotiator");
```

Comment:
Why is fallback negotiator is required? Is that by design?

Reply:
Yes, when the channel credentials are implemented and XdsCredentials are used then fallback creds is mandatory (which means the fallback negotiator is mandatory here).

Reply:
Thinking more about it, if a falback creds is not provided but the channel never needs to use it, it should not be an error. Let me think more about it.

Reply:
Yeah, sometimes the user does not want to use plaintext but can not think of any viable fallback negotiator, in that case if they can reply only primary, seems we should allow it and leave fallback negotiator unset and only fail when the primary does not work.

Reply:
fixed. PTAL